codesign/test/src/fake_google_cloud_storage.dart

```diff
@@ -0,0 +1,75 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:codesign/codesign.dart';
+import 'package:file/file.dart';
+import 'package:process/process.dart';
+
+/// Utility function class to handle upload/download of files from/to google cloud.
+class FakeGoogleCloudStorage extends GoogleCloudStorage {
+  FakeGoogleCloudStorage();
+
+  /// Wrapper function to upload code signed flutter engine artifact to google cloud bucket.
+  @override
+  Future<void> uploadEngineArtifact({
+    required String localPath,
+    required String remotePath,
+    required String commitHash,
+    required ProcessManager processManager,
+    int exitCode = 0,
+  }) async {
+    final String fullRemotePath = '$gsCloudBaseUrl/flutter/$commitHash/$remotePath';
+    return await fakeUploadFunction(
+      localPath: localPath,
+      destinationUrl: fullRemotePath,
+      processManager: processManager,
+      exitCode: exitCode,
+    );
+  }
+
+  /// Wrapper function to download flutter engine artifact from google cloud bucket.
+  @override
+  Future<File> downloadEngineArtifact({
+    required String remotePath,
+    required String localPath,
+    required String commitHash,
+    required ProcessManager processManager,
+    required Directory rootDirectory,
+    int exitCode = 0,
+  }) async {
+    final String sourceUrl = '$gsCloudBaseUrl/flutter/$commitHash/$remotePath';
+    return await fakeDownloadFunction(
+      sourceUrl: sourceUrl,
+      localPath: localPath,
+      processManager: processManager,
+      rootDirectory: rootDirectory,
+      exitCode: exitCode,
+    );
+  }
+
+  Future<void> fakeUploadFunction({
```

Comment:
These seem simple enough to inline in the original methods

Reply:
umm I removed this fake class, and reverted back to the fake processManager topology. 

umm I might be wrong, when inlining fake class into real class, I found it a bit messy to `if` check which class is real and provide exit code accordingly. So I removed the fake class.